src/ripple/shamap/SHAMapMissingNode.h

```diff
@@ -34,33 +36,44 @@ enum class SHAMapType
     FREE         = 3,    // A tree not part of a ledger
 };
 
+inline
+std::string
+to_string(SHAMapType t)
+{
+    if (t == SHAMapType::TRANSACTION)
+        return "Transaction Tree";
+    if (t == SHAMapType::STATE)
+        return "State Tree";
+    if (t == SHAMapType::FREE)
+        return "Free Tree";
+    return std::to_string(safe_cast<std::underlying_type_t<SHAMapType>>(t));
+}
+
 class SHAMapMissingNode
     : public std::runtime_error
 {
-private:
-    SHAMapType mType;
-    SHAMapHash mNodeHash;
-    uint256    mNodeID;
 public:
-    SHAMapMissingNode (SHAMapType t,
-                       SHAMapHash const& nodeHash)
-        : std::runtime_error ("SHAMapMissingNode")
-        , mType (t)
-        , mNodeHash (nodeHash)
+    SHAMapMissingNode (SHAMapType t, SHAMapHash const& hash)
```

Comment:
What do you think of adding the stack trace here when in debug mode?

Reply:
I thought about doing this in `LogThrow` actually.

Reply:
On Mac and Linux, the existing terminate handler creates a very good stack dump.  I think we used to have a custom terminate handlers and got rid of them in order to _improve_ our stack dumps.

Reply:
The question is *where* do we get the usable stack dumps from? I know on Mac you can get them in the log, but what about Linux?

Reply:
As not-a-Linux-user, I don't know.  I thought we had that knowledge within the rippled team though.  We used to get them mailed to us whenever a production server went down (hasn't happened in awhile).

Perhaps I'm misunderstanding this thread...

Reply:
That was caused by a script we were running on production servers ran by Ripple that would detect the crash and automatically ran `gdb` on the resulting core file. The question is what happens if there is no core file? On the Mac we do get a very pretty call stack in the logs. We'll have to research what happens on Linux.